torch/_decomp/decompositions.py

```diff
@@ -3350,11 +3403,14 @@ def upsample_bilinear2d(
     if input.device.type == "cuda" and n_channels < 16:
         memory_format = torch.contiguous_format
 
+    assert isinstance(result, torch.Tensor)
+
     result = result.contiguous(memory_format=memory_format)
 
     if not input.is_floating_point():
         result = result.round()
 
+    result = result.to(input.dtype)
```

Comment:
@vfdev-5, I added this to match the eager mode where the result has the same type as the input.

Reply:
I do not think this is needed.

Reply:
Here's a simple example where this is necessary.
```
import torch

def f(x):
    return torch.nn.Upsample(scale_factor=tuple([2.0, 3.0]), mode="bilinear")(x)
    
dtype = torch.uint8
input = torch.arange(1, 7, dtype=dtype).view(1, 1, 2, 3)

print(f(input).dtype)
print(torch.compile(f)(input).dtype)
```

Reply:
Is this a test anywhere? If not, can you add it to the relevant OpInfo?

Reply:
This was an issue in this PR where I misplaced an `pw_cast_for_opmath`. Removed now.

Reply:
There is a test already and it failed in this PR and is fixed now.